---
title: "Analysis Proposal"
author: "Sean Stuntz"
date: "January 17, 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Bioinformatics Automated Workflow System (BAWS)

BAWS allows users to quickly filter, visualize, and analyze taxonomic marker data.

## Why is it Useful?

The app is intended to, as the name describes, automate various data formatting, visualization and statistical procedures for 16s RNA genetic sequence taxonomic markers. In a nutshell, the app will generate data tables from a raw data set, assign taxonomy and create a phyloseq object. The phyloseq object can then be used for subsequent analysis or for sharing information between researchers. The intent is to take raw, genetic sequence data and format it or analyze it until it is able to be shared with the microbiome research community.

The app will enable automation for the following processes:

*	Referencing 16s RNA gene sequence raw data file, generate data table with user specified number of forward and reverse passes

*	Cluster sequences into Operational Taxonomic Units meeting a certain dissimilarity threshold

*	Construct sequence table and remove chimeras

* Assign taxonomic markers

* Construct a phylogenic tree

* Build Phyloseq object by consolidating information above

There is nothing novel being performed in the analysis performed within the app. All it is intended to offer is automation of these processes in order save microbiome researchers time when performing routine formatting and analyses. In fact, the app builds off of R packages commonly used in this community including phangorn, ggplot2, dada2, DESeq2, and structSSI. To benefit from this app, users need to have an understanding of genetic sequencing analysis techniques and basic statistics. Users can find the app at my GitHub repository located here.

## Feature Table

![](C:\Users\Corie & Sean\Documents\table.png)
